### ISSUE TITLE
Can't get existing credentials_gce() to work

### ISSUE THREAD

**Issue by wlongabaugh** [r-lib/gargle#97]
----
We are trying to use the `biqrquery` package on a Google VM, where we obtain the needed token from the VM metadata service, i.e. using just the function `credentials_gce()`. In our use case, minting service account keys is not an option. Despite trying various approaches, we were unable to get it to work, and would get the error messages:

```
Error: Can't get Google credentials.
Are you running bigrquery in a non-interactive session? Consider:
  * Call `bq_auth()` directly with all necessary specifics.
Execution halted
```

Here is an example that creates the above  messages:
```r
library(gargle)
library(bigrquery)
projectID = "my-cloud-project"
cred_funs_clear()
cred_funs_add(credentials_gce)
scopes = c("https://www.googleapis.com/auth/bigquery", "https://www.googleapis.com/auth/cloud-platform")
bq_auth(scopes)
```

We tracked the issue down into `gargle` and tweaked `credentials_gce()` a bit to get it working. We can submit a pull request if this might be useful? Thx.

**Comment by jennybc** [comment]
> We tracked the issue down into gargle and tweaked credentials_gce() a bit to get it working. We can submit a pull request if this might be useful? Thx.

That would be great! I have struggled to get feedback and specifics from people using / needing the GCE credentials flow, so I would love to hear from you, via PR or otherwise.

**Comment by wlongabaugh** [comment]
Thanks. I have made a pull request. I might not have used the best approaches for the three needed fixes, but I got it working. Happy to provide feedback on this use case. When working with BigQuery tables containing controlled access cancer variant data on the ISB-CGC Cancer Cloud Resource, service accounts that have been approved to access controlled data *cannot* have keys minted. Thus our need for this flow. Thanks for providing the framework!
